Add support for `inH₂O` pressure unit

ekobres opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--

-->
This change adds the HVAC industry standard unit of measurement for static pressure, inH2O (inches of water) which has a direct conversion to Pascals.

Modern HVAC systems can report system static pressure. They report it in inH2O - for example the Carrier Infinity system uses this unit for its static pressure sensor. This type of sensor can be used to monitor overall system health, detect anomalies, or track how restrictive an air filter becomes over time.

I propose adding "inH2O" as a new PRESSURE measurement unit.
I propose adding support for conversion to/from Pascals as other pressure units do and also to ensure users who may be tracking such sensors in non HVAC-standard units (such as PSI) can switch units and retain their history if desired.
I propose adding appropriate conversion tests to the `test_unit_conversion.py` tests.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
I am a longtime HA user, first time contributor. I have tried to follow the guidelines faithfully. Please be gentle.

I felt the best choice was new feature - but it does not immediately add new features to an integration. Rather, it gives integrations the option to use this new UOM rather than a generic UOM that can't be converted to other pressure units, or even expressed as pressure.
-->

- This PR fixes or closes issue: fixes #146997 
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/39994
- Link to developer documentation pull request: https://github.com/home-assistant/developers.home-assistant/pull/2723
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] I have followed the [perfect PR recommendations][perfect-pr]
- [X] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

ekobres committed:
Add inH2O unit support and tests for pressure conversion

ekobres committed:
Fix formatting: remove extraneous space in test_unit_conversion.py

ekobres commented on tests/util/test_unit_conversion.py:
```diff
@@ -629,26 +629,37 @@
+        (100, UnitOfPressure.INH2O, 3.6127291827353996, UnitOfPressure.PSI),
+        (100, UnitOfPressure.INH2O, 186.83201548767, UnitOfPressure.MMHG),
+        (100, UnitOfPressure.INH2O, 7.3555912463681, UnitOfPressure.INHG),
+        (100, UnitOfPressure.INH2O, 24908.890833333 , UnitOfPressure.PA),
```
Updated.

epenet reviewed (dismissed):
1 - there is no reason to delete the vscode files - please revert that change
2 - please expand the PR description to indicate why you feel the new unit is needed, and which integrations would make use of it
3 - what does `249.0889083333348` correspond to as a number? could it be a named constant so we know what it is?
4 - please ensure that docstrings are adjusted in `sensor` and `number` platforms for pressure device class
5 - please ensure that developer docs are updated for pressure device class in `sensor` and `number` platforms

ekobres committed:
Restore deleted VSCode config files

ekobres committed:
Add support for inH2O unit of measurement in
number and sensor components. Add constant for
inH2O to Pascal conversion factor in
unit_conversion.py.

ekobres commented:
> 1 - there is no reason to delete the vscode files - please revert that change 
2 - please expand the PR description to indicate why you feel the new unit is needed, and which integrations would make use of it 
3 - what does `249.0889083333348` correspond to as a number? could it be a named constant so we know what it is? 
4 - please ensure that docstrings are adjusted in `sensor` and `number` platforms for pressure device class 
5 - please ensure that developer docs are updated for pressure device class in `sensor` and `number` platforms

Thanks for the review. 

1 - Done - sorry missed this.
2 - Done - somehow I had written the whole thing in a comment block.
3 - Done - The magic number here is the SI (Pascals) conversion factor. (How many inH2O are in one PA - which the unit conversion utility uses to convert all pressure units of measure.) It serves the same purpose for inH2O as [this line](https://github.com/ekobres/core/blob/100d5f08d8cb5eb7d94097bdd7bf7e6d3eda0b96/homeassistant/util/unit_conversion.py#L420) serves for PSI. Given the existing PSI conversion factor uses this format, I took the path to use the equivalent approach of those who have gone before. I have made the change and added _INH2O_TO_PA as a constant.
4 - Done
5 - Happy to do this. I will open a PR there for the changes.

joostlek commented:
Is there an architectural proposal for this?

ekobres commented:
> Is there an architectural proposal for this?

This PR adds a new unit of measurement (inH₂O) to the core unit registry. It doesn’t modify any existing units or conversions and has zero impact on current integrations.

While it does touch a shared core component, the change is purely additive and enables future integrations (particularly in HVAC and industrial contexts) to report pressure using a more appropriate and widely used unit without resorting to workarounds.

Since there’s no change in behavior and no potential for breakage, I don’t believe this warrants a full architectural proposal. That said, I’m happy to open one if you think it's necessary.

emontnemery reviewed (commented):
This looks good, but I think we should do "inH₂O" instead of "inH2O"

emontnemery committed:
Update homeassistant/const.py

emontnemery committed:
Apply suggestions from code review

emontnemery committed:
Update homeassistant/util/unit_conversion.py

ekobres committed:
tests: sensor/recorder: include inH₂O in supported pressure units

ekobres committed:
Added US/Metric conversions.

emontnemery reviewed (approved):
Thanks, @ekobres 👍
